langwatch_nlp/langwatch_nlp/logger.py

```
@@ -1,20 +1,163 @@
+    return new_dict
+
+
+def _configure_structlog() -> None:
+    """Configure structlog based on environment."""
+    is_test = os.environ.get("NODE_ENV") == "test" or os.environ.get("PYTEST_CURRENT_TEST")
+    is_prod = os.environ.get("NODE_ENV") == "production"
+    log_level_str = os.environ.get("LOG_LEVEL", "INFO" if not is_test else "ERROR")
+    log_level = getattr(logging, log_level_str.upper(), logging.INFO)
+
+    def add_logger_name(
+        logger: WrappedLogger, method_name: str, event_dict: EventDict
```

Ruff (0.14.14)

[warning] Unused function argument: `logger`

(ARG001)